model/patron.py

```diff
@@ -300,6 +282,188 @@ def synchronize_annotations(self, value):
+            # configuration problem, but we shouldn't make the patron
+            # pay the price -- just pick the first one.
+            logging.error(
+                "Multiple root lanes found for patron type %s.",
+                self.external_type
+            )
+        return lanes[0]
+
+    def work_is_age_appropriate(self, work_audience, work_target_age):
+        """Is the given audience and target age an age-appropriate match for this Patron?
+
+        NOTE: What "age-appropriate" means depends on some policy questions
+        that have not been answered and may be library-specific. For
+        now, it is determined by comparing audience and target age to that of the
+        Patron's root lane.
+
+        This is designed for use when reasoning about works in
+        general. If you have a specific Work in mind, use
+        `Work.age_appropriate_for_patron`.
+
+        :param work_audience: One of the audience constants from
+           Classifier, representing the general reading audience to
+           which a putative work belongs.
+
+        :param work_target_age: A number or 2-tuple representing the target age
+           or age range of a putative work.
+
+        :return: A boolean
+
+        """
+        root = self.root_lane
+        if not root:
+            # The patron has no root lane. They can interact with any
+            # title.
+            return True
+
+        # The patron can interact with a title if any of the audiences
+        # in their root lane (in conjunction with the root lane's target_age)
+        # are a match for the title's audience and target age.
+        return any(
+            self.age_appropriate_match(
+                work_audience, work_target_age,
+                audience, root.target_age
+            )
+            for audience in root.audiences
+        )
+
+    @classmethod
+    def age_appropriate_match(
+        cls, work_audience, work_target_age,
+        reader_audience, reader_age
+    ):
+        """Match the audience and target age of a work with that of a reader,
+        and see whether they are an age-appropriate match.
+
+        NOTE: What "age-appropriate" means depends on some policy
+        questions that have not been answered and may be
+        library-specific. For now, non-children's books are
+        age-inappropriate for young children, and children's books are
+        age-inappropriate for children too young to be in the book's
+        target age range.
+
+        :param reader_audience: One of the audience constants from
+           Classifier, representing the general reading audience to
+           which the reader belongs.
+
+        :param reader_age: A number or 2-tuple representing the age or
+           age range of the reader.
+        """
+        if reader_audience is None:
+            # A patron with no particular audience restrictions
+            # can see everything.
+            return True
```

Comment:
Would it make sense to add logs with the `DEBUG` level so later it would be easier to troubleshoot this code?

Reply:
Good idea.